### DISCUSSION TITLE
PWRMON for Asus-Merlin FW Routers

### DISCUSSION THREAD

**Discussion by ViktorJp** [jasonacox/Powerwall-Dashboard#136] [Show and tell]
Not exactly on the same level as your amazing Powerwall-Dashboard project, but still really happy with the result, and wanted to show and tell, even if it gives you a chuckle.  PWRMON is a shell script that runs off Asus-brand routers - none of this fancy grafana, influxdb's -- no, no.... this is just a real-time tap into your PW gateway to provide as-is stats.  Meant to be displayed on a cheap tablet, or wall-mounted display through crude SSH windows, this will provide you with some options to view live power flows (painfully reconstructed into fancy-lookin' ASCII), ingress and egress power stats coming to/from solar, grid, home and batteries, as well as a large font dashboard meant to be read from across the room in order to give you current info at a glance.  Enjoy!  ;)

![pwrmon-1 25-pg1](https://user-images.githubusercontent.com/97465574/202908151-d2f3978d-b068-47cb-9e4a-01ef170f7a12.jpg)

![pwrmon-1 25-pg2](https://user-images.githubusercontent.com/97465574/202908157-9a6cfc5f-c6e2-4c9b-990f-3fa2a146e310.jpg)

![pwrmon-1 25-pg3](https://user-images.githubusercontent.com/97465574/202908172-731df88e-f25b-4411-b25e-ecd34a9e1e9e.jpg)

More info here:
https://www.snbforums.com/threads/pwrmon-v1-25-nov-20-2022-monitor-your-tesla-powerwall-s-solar-grid-home-stats-new-weather-sms-email-notifications-large-dashboard.80203/

Github:
https://github.com/ViktorJp/PWRMON

**Comment by jasonacox** [comment]
Wow!  This is absolutely brilliant @ViktorJp !  I love it! ❤️ 😂  

What would it take to port it to a generic Linux / MacOS host?  I did a quick browse through your shell code.  Very nice.  I have to admit, I want to see those ASCII power flows. 😁

**Reply by ViktorJp** [reply]
Hahah :smirk: I wish I knew how... it's pretty tight with the Asus-Merlin Firmware capabilities/functionality running on compatible Asus routers, so I'm not sure how forgiving it would be on other platforms... You could always find a nice Black Friday deal for a [compatible router](https://www.asuswrt-merlin.net/about), or find a good excuse to upgrade that ol' home router for Xmas, and explore the vast custom development efforts happening on the Merlin-Asus Firmware out there.  It's been a lot of fun these past many years...  :grin:
